components/StyledRadioList.js

```diff
@@ -62,30 +70,20 @@ const StyledRadioList = ({
   labelProps,
   ...props
 }) => {
-  const [selected, setSelected] = useState(props.defaultValue);
+  const [localStateSelected, setSelected] = useState(getValueFromProps(props));
   const keyExtractor = getKeyExtractor(options, keyGetter);
   const items = getItems(options, keyExtractor);
   const defaultValueStr = props.defaultValue !== undefined && props.defaultValue.toString();
+  const valueFromProps = getValueFromProps(props);
+  const selected = isUndefined(valueFromProps) ? localStateSelected : valueFromProps;
+  const checkedItem = isUndefined(props.value) ? localStateSelected : selected;
```

Comment:
The local state is not useful if we use `value`, so we can simplify that further by removing `getValueFromProps` and refactoring like:
```suggestion
  const [localStateSelected, setSelected] = useState(props.defaultValue);
  const keyExtractor = getKeyExtractor(options, keyGetter);
  const items = getItems(options, keyExtractor);
  const defaultValueStr = props.defaultValue !== undefined && props.defaultValue.toString();
  const checkedItem = !isUndefined(props.value) ? props.value : localStateSelected;
```

(see https://github.com/opencollective/opencollective-frontend/pull/4454/commits/7a85e9768e6d497e52db9bf22d563722329224ad)